### PR TITLE
NEW clean values and amount in FEC import

### PR DESCRIPTION
**Pull request by lvessiller-opendsi**
NEW clean values and amount in FEC import
- uses floatval for amounts (debit, credit and multicurrency_amount) to insert in accounting bookeeping table
- uses trim method for "piece_num" field to insert in accounting bookeeping table

### CODE REVIEW

**Comment by lvessiller-opendsi** [comment]
@eldy 
I have Travis error but it seems not to be linked with this fix : 
The command "~/.phpenv/versions/$(phpenv version-name)/sbin/php-fpm" failed and exited with 127 during

File: htdocs/accountancy/class/accountancyimport.class.php [hunk 1]
```diff
@@ -55,21 +85,26 @@ public function computeAmount(&$arrayrecord, $fieldname, &$listfields, &$listval
...
+		$debit_val = trim($arrayrecord[$debit_index]['val']);
+		$credit_val = trim($arrayrecord[$credit_index]['val']);
+		$debit  = floatval($debit_val);
+		$credit = floatval($credit_val);
 		if (!empty($debit)) {
 			$amount = $debit;
 		} else {
 			$amount = $credit;
 		}
+		$listvalues[$debit_index] = "'" . $debit . "'";
```
Comment (eldy) [inline htdocs/accountancy/class/accountancyimport.class.php:99]: Can you test with this code instead
`$listvalues[$debit_index] = "'" . $this->db->escape($debit) . "'";`

File: htdocs/core/modules/import/import_csv.modules.php [hunk 2]
```diff
@@ -612,7 +612,7 @@ public function import_insert($arrayrecord, $array_match_file_to_database, $obji
 										break;
 									}
 									$classinstance = new $class($this->db);
-									$res = call_user_func_array(array($classinstance, $method), array(&$arrayrecord));
+									$res = call_user_func_array(array($classinstance, $method), array(&$arrayrecord, $fieldname, &$listfields, &$listvalues, $key - 1));
```
Comment (eldy) [inline htdocs/core/modules/import/import_csv.modules.php:615]: Why do you need the &  on $listfields    and   $listvalues  ?
Only  $arrayrecord  (the array that would be used for output should be modified).   $listfield and $listvalues are entry and must remained unchanged during all exposed or you expose the code to bug by having the var modified for one call of method and for another call, input data will not be the same.

File: htdocs/accountancy/class/accountancyimport.class.php [hunk 3]
```diff
@@ -55,21 +85,26 @@ public function computeAmount(&$arrayrecord, $fieldname, &$listfields, &$listval
 			$fieldname = $fieldArr[1];
 		}
 
-		$debit  = floatval(trim($arrayrecord[11]['val']));
-		$credit = floatval(trim($arrayrecord[12]['val']));
+		$debit_index = 11;
```
Comment (eldy) [inline htdocs/accountancy/class/accountancyimport.class.php:88]: This method is used by the import process. The field debit will not always be at position 11.

**Comment by lvessiller-opendsi** [comment]
@eldy 
I reviewed this PR with your suggests and resolved all conflicts.
Could you see to merge it ?

**Comment by eldy** [comment]
Sorry i can't merge this PR. As i explained, there is major conception pb in this PR. This my previous comment.

**Comment by lvessiller-opendsi** [comment]
@eldy
I changed files and integrated all your comments

File: htdocs/accountancy/class/accountancyimport.class.php [hunk 4]
```diff
@@ -73,17 +108,24 @@ public function computeAmount(&$arrayrecord, $fieldname, &$listfields, &$listval
...
+
+			$debit  = floatval($arrayrecord[$debit_index]['val']);
+			$credit = floatval($arrayrecord[$credit_index]['val']);
+			if (!empty($debit)) {
+				$amount = $debit;
+			} else {
+				$amount = $credit;
+			}
+
+			$listfields[] = $fieldname;
```
Comment (eldy) [inline htdocs/accountancy/class/accountancyimport.class.php:125]: You should not modify content of $listfields and $listvalues. Only the $arrayrecord that is the array that will be used for output should be modified.
Modification of a value should be done into the $arrayrecord directly. You should know which entry to modify by using the ($key - 1) parameter that is send when you call
`$res = call_user_func_array(array($classinstance, $method), array(&$arrayrecord, $fieldname, $listfields, $listvalues, $key - 1));`
but that is missing in parameters of computeAmount().

**Comment by eldy** [comment]
@lvessiller-opendsi I merged the PR but i have to made some changes to it. Can you try the result is still working ?

**Pull request merged**